Video game analysis

In [66]:
import os

%load_ext sql
database_url = os.getenv('videogame_sales')
%sql $database_url

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [67]:
%%sql

WITH top_critic_years AS(
    SELECT EXTRACT(YEAR FROM release_date) AS year, 
        COUNT(title) AS num_games_reviewed,
        ROUND(AVG(critic_score), 2) AS avg_critic_score
    FROM dim_product AS dp
    LEFT JOIN dim_review AS dr
    ON dp.unique_title = dr.unique_title
    WHERE critic_score IS NOT NULL
    GROUP BY year
    ORDER BY avg_critic_score DESC
    LIMIT 10
), top_critic_years_often_reviewed AS(
    SELECT EXTRACT(YEAR FROM release_date) AS year, 
        COUNT(title) AS num_games_reviewed, 
        ROUND(AVG(critic_score),2) AS avg_critic_score
    FROM dim_product dp
    INNER JOIN dim_review dr
    ON dp.unique_title = dr.unique_title
    WHERE critic_score IS NOT NULL
    GROUP BY EXTRACT(YEAR FROM release_date)
    HAVING COUNT(DISTINCT title) > 30
    ORDER BY avg_critic_score DESC
    LIMIT 10
)
SELECT year, 
    num_games_reviewed,
    avg_critic_score
FROM top_critic_years
EXCEPT
SELECT year, 
    num_games_reviewed,
    avg_critic_score
FROM top_critic_years_often_reviewed
ORDER BY avg_critic_score DESC;

 * postgresql://postgres:***@localhost:5432/Videogame_Sales
9 rows affected.


[(Decimal('1985'), 1, Decimal('10.00')),
 (Decimal('1992'), 4, Decimal('9.13')),
 (Decimal('1987'), 1, Decimal('9.00')),
 (Decimal('1991'), 6, Decimal('8.80')),
 (Decimal('1993'), 12, Decimal('8.37')),
 (Decimal('1994'), 15, Decimal('8.26')),
 (Decimal('1988'), 1, Decimal('8.00')),
 (Decimal('2020'), 13, Decimal('7.95')),
 (Decimal('2023'), 2, Decimal('7.80'))]

All years with very high scores only had a few reviews and therefore were easy outliers.

1. Analyze market share of platforms for Total Sales

In [68]:
%%sql

SELECT platform,
    SUM(na_sales) AS North_america,
    SUM(jp_sales) AS Japan,
    SUM(other_sales) AS other,
    SUM(pal_sales) AS Europe_Africa,
    SUM(total_sales) AS Total
FROM fact_sales AS fs 
LEFT JOIN dim_product AS dp 
ON dp.unique_title = fs.unique_title
WHERE total_sales IS NOT NULL
GROUP BY platform
ORDER BY SUM(total_sales) DESC

 * postgresql://postgres:***@localhost:5432/Videogame_Sales
9 rows affected.


[('PlayStation', Decimal('1395.37'), Decimal('349.20'), Decimal('431.27'), Decimal('1065.01'), Decimal('3240.53')),
 ('Nintendo', Decimal('923.91'), Decimal('252.15'), Decimal('92.84'), Decimal('356.26'), Decimal('1628.37')),
 ('Xbox', Decimal('692.01'), Decimal('10.51'), Decimal('101.14'), Decimal('323.50'), Decimal('1127.23')),
 ('Other', Decimal('168.68'), Decimal('20.79'), Decimal('8.31'), Decimal('56.61'), Decimal('254.93')),
 ('PC', Decimal('58.64'), None, Decimal('13.50'), Decimal('95.96'), Decimal('168.95')),
 ('Atari', Decimal('69.59'), None, Decimal('0.68'), Decimal('4.19'), Decimal('74.56')),
 ('Sega', Decimal('17.77'), Decimal('41.88'), Decimal('0.88'), Decimal('5.40'), Decimal('65.94')),
 ('NEC', None, Decimal('0.19'), None, None, Decimal('0.19')),
 ('Mobile', None, None, Decimal('0.0'), Decimal('0.02'), Decimal('0.03'))]

In [69]:
%%sql

WITH top3_sales AS(
    SELECT SUM(total_sales) AS Total_top3
    FROM fact_sales AS fs 
    LEFT JOIN dim_product AS dp 
    ON dp.unique_title = fs.unique_title
    WHERE total_sales IS NOT NULL
    AND platform IN ('PlayStation', 'Xbox', 'Nintendo')
    ORDER BY SUM(total_sales) DESC
), 
other_sales AS(
    SELECT SUM(total_sales) AS Total_other
    FROM fact_sales AS fs 
    LEFT JOIN dim_product AS dp 
    ON dp.unique_title = fs.unique_title
    WHERE total_sales IS NOT NULL
    AND platform NOT IN ('PlayStation', 'Xbox', 'Nintendo')
    ORDER BY SUM(total_sales) DESC
)
SELECT ROUND(Total_top3 / ( Total_top3 + Total_other),2) AS CR3
FROM top3_sales, other_sales



 * postgresql://postgres:***@localhost:5432/Videogame_Sales
1 rows affected.


[(Decimal('0.91'),)]

This shows the market share of PlayStation, Nintendo and Xbox is 91% of the total market share, therefore focus of the analysis will be on these platforms only from hereon.

2. Analyze percentage of sales per region

In [70]:
%%sql

SELECT platform,
    ROUND(SUM(na_sales) / SUM(total_sales),2) AS North_america,
    ROUND(SUM(jp_sales) / SUM(total_sales),2) AS Japan,
    ROUND(SUM(other_sales) / SUM(total_sales),2) AS other,
    ROUND(SUM(pal_sales) / SUM(total_sales),2) AS Europe_Africa,
    ROUND(SUM(total_sales) / SUM(total_sales),2) AS Total
FROM fact_sales AS fs 
LEFT JOIN dim_product AS dp 
ON dp.unique_title = fs.unique_title
WHERE total_sales IS NOT NULL
AND platform IN ('PlayStation', 'Xbox', 'Nintendo')
GROUP BY platform
ORDER BY SUM(total_sales) DESC

 * postgresql://postgres:***@localhost:5432/Videogame_Sales
3 rows affected.


[('PlayStation', Decimal('0.43'), Decimal('0.11'), Decimal('0.13'), Decimal('0.33'), Decimal('1.00')),
 ('Nintendo', Decimal('0.57'), Decimal('0.15'), Decimal('0.06'), Decimal('0.22'), Decimal('1.00')),
 ('Xbox', Decimal('0.61'), Decimal('0.01'), Decimal('0.09'), Decimal('0.29'), Decimal('1.00'))]

Note: Xbox has a very low market share in Japan and is highly focused on just North America compared to the others.

3. Analyze top 5 selling genres per platform based on Total sales

In [71]:
%%sql

WITH RankedGenres AS (
    SELECT genre, platform, 
           SUM(total_sales) AS total_sales,
           ROW_NUMBER() OVER (PARTITION BY platform ORDER BY SUM(total_sales) DESC) AS rank
    FROM fact_sales AS fs
    LEFT JOIN dim_product AS dp 
    ON dp.unique_title = fs.unique_title
    WHERE total_sales IS NOT NULL
    AND platform IN ('PlayStation', 'Xbox', 'Nintendo')
    GROUP BY genre, platform
)
SELECT platform, genre, rank, total_sales
FROM RankedGenres
WHERE rank <= 5
ORDER BY platform, rank;


 * postgresql://postgres:***@localhost:5432/Videogame_Sales
15 rows affected.


[('Nintendo', 'Misc', 1, Decimal('248.86')),
 ('Nintendo', 'Action', 2, Decimal('234.95')),
 ('Nintendo', 'Sports', 3, Decimal('222.20')),
 ('Nintendo', 'Platform', 4, Decimal('156.40')),
 ('Nintendo', 'Simulation', 5, Decimal('147.88')),
 ('PlayStation', 'Sports', 1, Decimal('687.77')),
 ('PlayStation', 'Action', 2, Decimal('587.88')),
 ('PlayStation', 'Shooter', 3, Decimal('499.82')),
 ('PlayStation', 'Racing', 4, Decimal('318.91')),
 ('PlayStation', 'Misc', 5, Decimal('221.15')),
 ('Xbox', 'Shooter', 1, Decimal('321.67')),
 ('Xbox', 'Action', 2, Decimal('208.01')),
 ('Xbox', 'Sports', 3, Decimal('189.89')),
 ('Xbox', 'Racing', 4, Decimal('81.15')),
 ('Xbox', 'Role-Playing', 5, Decimal('79.66'))]

Sales top 5 genres (based on total sales) | region

In [72]:
%%sql

SELECT genre,
    ROUND(SUM(na_sales) / SUM(total_sales), 2) AS North_America,
    ROUND(SUM(jp_sales) / SUM(total_sales), 2) AS Japan,
    ROUND(SUM(other_sales) / SUM(total_sales), 2) AS Other,
    ROUND(SUM(pal_sales) / SUM(total_sales), 2) AS Europe_Africa,
    ROUND(SUM(total_sales) / SUM(total_sales), 2) AS Total
FROM fact_sales AS fs 
LEFT JOIN dim_product AS dp 
ON dp.unique_title = fs.unique_title
WHERE total_sales IS NOT NULL
AND platform IN ('PlayStation', 'Xbox', 'Nintendo')
GROUP BY genre
ORDER BY SUM(total_sales) DESC
LIMIT 5;



 * postgresql://postgres:***@localhost:5432/Videogame_Sales
5 rows affected.


[('Sports', Decimal('0.51'), Decimal('0.09'), Decimal('0.11'), Decimal('0.29'), Decimal('1.00')),
 ('Action', Decimal('0.51'), Decimal('0.08'), Decimal('0.11'), Decimal('0.31'), Decimal('1.00')),
 ('Shooter', Decimal('0.52'), Decimal('0.03'), Decimal('0.12'), Decimal('0.33'), Decimal('1.00')),
 ('Misc', Decimal('0.53'), Decimal('0.10'), Decimal('0.11'), Decimal('0.26'), Decimal('1.00')),
 ('Racing', Decimal('0.51'), Decimal('0.03'), Decimal('0.11'), Decimal('0.35'), Decimal('1.00'))]

Note: Shooter game sales are relatively low in japan compared to the total.

4. Analyze the share of sales for each genre per region.

In [73]:
%%sql

WITH total AS (
    SELECT SUM(jp_sales) AS jp_total,
           SUM(na_sales) AS na_total,
           SUM(pal_sales) AS pal_total,
           SUM(other_sales) AS other_total,
           SUM(total_sales) AS total_total
    FROM fact_sales
)
SELECT genre,
    ROUND(SUM(na_sales) / (SELECT na_total FROM total), 2) AS North_America,
    ROUND(SUM(jp_sales) / (SELECT jp_total FROM total), 2) AS Japan,
    ROUND(SUM(other_sales) / (SELECT other_total FROM total), 2) AS Other,
    ROUND(SUM(pal_sales) / (SELECT pal_total FROM total), 2) AS Europe_Africa,
    ROUND(SUM(total_sales) / (SELECT total_total FROM total), 2) AS Total
FROM fact_sales AS fs 
LEFT JOIN dim_product AS dp 
ON dp.unique_title = fs.unique_title
WHERE total_sales IS NOT NULL
AND platform IN ('PlayStation', 'Xbox', 'Nintendo')
GROUP BY genre
ORDER BY SUM(total_sales) DESC
LIMIT 5;

 * postgresql://postgres:***@localhost:5432/Videogame_Sales
5 rows affected.


[('Sports', Decimal('0.17'), Decimal('0.15'), Decimal('0.19'), Decimal('0.17'), Decimal('0.17')),
 ('Action', Decimal('0.16'), Decimal('0.12'), Decimal('0.17'), Decimal('0.17'), Decimal('0.16')),
 ('Shooter', Decimal('0.14'), Decimal('0.04'), Decimal('0.16'), Decimal('0.15'), Decimal('0.13')),
 ('Misc', Decimal('0.09'), Decimal('0.08'), Decimal('0.09'), Decimal('0.07'), Decimal('0.08')),
 ('Racing', Decimal('0.07'), Decimal('0.02'), Decimal('0.08'), Decimal('0.09'), Decimal('0.07'))]

Note: The Shooter and Racing genre only makes up for 4% and 2% in the Japanese game market respectively. This is very low compared to other regions.

5. Check top 5 genre in Japan

In [74]:
%%sql

WITH total AS (
    SELECT SUM(jp_sales) AS jp_total
    FROM fact_sales
)
SELECT genre,
    ROUND(SUM(jp_sales) / (SELECT jp_total FROM total), 2) AS Japan
FROM fact_sales AS fs 
LEFT JOIN dim_product AS dp 
ON dp.unique_title = fs.unique_title
WHERE jp_sales IS NOT NULL
AND platform IN ('PlayStation', 'Xbox', 'Nintendo')
GROUP BY genre
ORDER BY SUM(jp_sales) DESC
LIMIT 5;

 * postgresql://postgres:***@localhost:5432/Videogame_Sales
5 rows affected.


[('Role-Playing', Decimal('0.18')),
 ('Sports', Decimal('0.15')),
 ('Action', Decimal('0.12')),
 ('Misc', Decimal('0.08')),
 ('Fighting', Decimal('0.07'))]

6. Analyze top 5 selling genres per platform based on Japanese sales (compare to 3)

In [75]:
%%sql

WITH RankedGenres AS (
    SELECT genre, platform, 
           SUM(jp_sales) AS Japan,
           ROW_NUMBER() OVER (PARTITION BY platform ORDER BY SUM(jp_sales) DESC) AS rank
    FROM fact_sales AS fs
    LEFT JOIN dim_product AS dp 
    ON dp.unique_title = fs.unique_title
    WHERE jp_sales IS NOT NULL
    AND platform IN ('PlayStation', 'Xbox', 'Nintendo')
    GROUP BY genre, platform
)
SELECT platform, genre, rank, Japan
FROM RankedGenres
WHERE rank <= 5
ORDER BY platform, rank;

 * postgresql://postgres:***@localhost:5432/Videogame_Sales
15 rows affected.


[('Nintendo', 'Role-Playing', 1, Decimal('50.72')),
 ('Nintendo', 'Sports', 2, Decimal('37.24')),
 ('Nintendo', 'Misc', 3, Decimal('26.42')),
 ('Nintendo', 'Platform', 4, Decimal('21.82')),
 ('Nintendo', 'Puzzle', 5, Decimal('19.57')),
 ('PlayStation', 'Role-Playing', 1, Decimal('65.83')),
 ('PlayStation', 'Sports', 2, Decimal('60.62')),
 ('PlayStation', 'Action', 3, Decimal('56.43')),
 ('PlayStation', 'Fighting', 4, Decimal('27.68')),
 ('PlayStation', 'Misc', 5, Decimal('26.02')),
 ('Xbox', 'Shooter', 1, Decimal('2.65')),
 ('Xbox', 'Action', 2, Decimal('2.40')),
 ('Xbox', 'Role-Playing', 3, Decimal('1.91')),
 ('Xbox', 'Fighting', 4, Decimal('0.74')),
 ('Xbox', 'Racing', 5, Decimal('0.57'))]

Note: Xbox has very similar highest selling genres in (3) and (6), whereas PlayStation and Nintendo have much different top-selling genres. Shooter and Racing are both in the top 5 genres sold in total sales for PlayStation and Nintendo, but dropped in the Japanese top 5 genres.
The top-selling genre for both PlayStation and Nintendo is Role-Playing, whereas this was not in the top 5 sold genres in (3). Note Role-playing is also the most sold-genre in Japan (see 5.) and racing and shooter games are unpopular (see 4.) 

7. Analyze the share of sales for the top 3 Japanese genres and Shooter & Racing genre for the total sales and Japanese sales by platform.

In [76]:
%%sql

WITH total AS (
    SELECT dp.platform,
           SUM(fs.total_sales) AS platform_total,
           SUM(fs.jp_sales) AS platform_jp_total
    FROM fact_sales fs
    LEFT JOIN dim_product dp 
    ON dp.unique_title = fs.unique_title
    WHERE fs.total_sales IS NOT NULL
    AND dp.platform IN ('PlayStation', 'Xbox', 'Nintendo')
    GROUP BY dp.platform
)
SELECT dp.platform, 
       dp.genre,
       ROUND(SUM(fs.total_sales), 2) AS genre_sales,
       ROUND((SUM(fs.total_sales) / t.platform_total) * 100, 2) AS total_genre_share,
       ROUND((SUM(fs.jp_sales) / t.platform_jp_total) * 100, 2) AS japan_genre_share
FROM fact_sales fs
LEFT JOIN dim_product dp 
ON dp.unique_title = fs.unique_title
LEFT JOIN total t
ON dp.platform = t.platform
WHERE fs.total_sales IS NOT NULL
AND dp.platform IN ('PlayStation', 'Xbox', 'Nintendo')
AND dp.genre IN ('Role-Playing', 'Sports', 'Action', 'Shooter', 'Racing')
GROUP BY dp.platform, dp.genre, t.platform_total, t.platform_jp_total
ORDER BY dp.platform, total_genre_share DESC;


 * postgresql://postgres:***@localhost:5432/Videogame_Sales
15 rows affected.


[('Nintendo', 'Action', Decimal('234.95'), Decimal('14.43'), Decimal('7.46')),
 ('Nintendo', 'Sports', Decimal('222.20'), Decimal('13.65'), Decimal('14.77')),
 ('Nintendo', 'Role-Playing', Decimal('98.08'), Decimal('6.02'), Decimal('20.12')),
 ('Nintendo', 'Racing', Decimal('84.50'), Decimal('5.19'), Decimal('1.40')),
 ('Nintendo', 'Shooter', Decimal('62.72'), Decimal('3.85'), Decimal('2.90')),
 ('PlayStation', 'Sports', Decimal('687.77'), Decimal('21.22'), Decimal('17.36')),
 ('PlayStation', 'Action', Decimal('587.88'), Decimal('18.14'), Decimal('16.16')),
 ('PlayStation', 'Shooter', Decimal('499.82'), Decimal('15.42'), Decimal('5.55')),
 ('PlayStation', 'Racing', Decimal('318.91'), Decimal('9.84'), Decimal('3.57')),
 ('PlayStation', 'Role-Playing', Decimal('209.55'), Decimal('6.47'), Decimal('18.85')),
 ('Xbox', 'Shooter', Decimal('321.67'), Decimal('28.54'), Decimal('25.21')),
 ('Xbox', 'Action', Decimal('208.01'), Decimal('18.45'), Decimal('22.84')),
 ('Xbox', 'Sports', Decimal('189.89'), Decimal('16.85'), Decimal('4.09')),
 ('Xbox', 'Racing', Decimal('81.15'), Decimal('7.20'), Decimal('5.42')),
 ('Xbox', 'Role-Playing', Decimal('79.66'), Decimal('7.07'), Decimal('18.17'))]

Noteworthy, PlayStation sells relatively much less shooter games in Japan (5.55%) compared to the share of shooter games they sell in total (15.42%). Furthermore, their share of sales for Role-playing games increases from 6.47% to 18.85% for total sales and Japanese sales respectively. Note, Role-playing genre was not in the top 5 most sold genre in total sales for PlayStation nor Japan. This implies PlayStation adjusts to the Japanese preferences. We see similar effects, though less in magnitude, for Nintendo.

For Xbox we see little adjustments in their share sales per genre. The shooter genre decreases slightly from 28.54% to 25.21% and the racing genre decreases from 7.2% to 5.42%. The share of Role-Playing games does increase substantially from 7.07% to 18.17%.

These findings substantiate the findings in (6).

PART 2: Sales per Game generation

In [77]:
%%sql

SELECT ROW_NUMBER() OVER(PARTITION BY sort_generation  ORDER BY SUM(total_sales) DESC) AS rank,
    platform,
    SUM(total_sales) AS Total_sales,
    game_generation
FROM Fact_sales fs
LEFT JOIN Dim_Product dp
ON dp.unique_title = fs.unique_title
WHERE total_sales IS NOT NULL
AND sort_generation > 3
AND sort_generation < 9
GROUP BY platform, sort_generation, game_generation
ORDER BY sort_generation DESC, total_sales DESC;



 * postgresql://postgres:***@localhost:5432/Videogame_Sales
15 rows affected.


[(1, 'PlayStation', Decimal('539.30'), 'Eighth Generation (2012–Present)'),
 (2, 'Xbox', Decimal('268.30'), 'Eighth Generation (2012–Present)'),
 (3, 'Nintendo', Decimal('134.71'), 'Eighth Generation (2012–Present)'),
 (1, 'PlayStation', Decimal('1145.58'), 'Seventh Generation (2005–2017)'),
 (2, 'Nintendo', Decimal('948.71'), 'Seventh Generation (2005–2017)'),
 (3, 'Xbox', Decimal('858.73'), 'Seventh Generation (2005–2017)'),
 (1, 'PlayStation', Decimal('1014.83'), 'Sixth Generation (1998–2013)'),
 (2, 'Nintendo', Decimal('119.53'), 'Sixth Generation (1998–2013)'),
 (3, 'Sega', Decimal('13.22'), 'Sixth Generation (1998–2013)'),
 (1, 'PlayStation', Decimal('537.01'), 'Fifth Generation (1993–2006)'),
 (2, 'Nintendo', Decimal('92.44'), 'Fifth Generation (1993–2006)'),
 (3, 'Sega', Decimal('33.48'), 'Fifth Generation (1993–2006)'),
 (1, 'Nintendo', Decimal('65.57'), 'Fourth Generation (1987–2004)'),
 (2, 'Sega', Decimal('19.20'), 'Fourth Generation (1987–2004)'),
 (3, 'Other', Decimal('0.10'), 'Fourth Generation (1987–2004)')]

Ever since the fifth generation PlayStation is the market leader. Noticeable is the large increase in share of Nintendo in the Seventh Generation which can be mainly attributed to the Wii; and the drop in sales from the 7th to the 8th generation. 

Furthermore, the rapid market entry of Xbox is also noticeable in the 7th generation which can be likely be explained to the succesfull launch of Xbox 360.

PART 3: Analysis top 10 highest reviewed years

Analyze top 10 highest reviewd years by critics


In [78]:
%%sql

SELECT EXTRACT(YEAR FROM release_date) AS year, 
    ROUND(AVG(critic_score), 2) AS avg_critic_score
FROM dim_product AS dp
LEFT JOIN dim_review AS dr
ON dp.unique_title = dr.unique_title
WHERE critic_score IS NOT NULL
GROUP BY year
ORDER BY avg_critic_score DESC
LIMIT 10;

 * postgresql://postgres:***@localhost:5432/Videogame_Sales
10 rows affected.


[(Decimal('1985'), Decimal('10.00')),
 (Decimal('1992'), Decimal('9.13')),
 (Decimal('1987'), Decimal('9.00')),
 (Decimal('1991'), Decimal('8.80')),
 (Decimal('1993'), Decimal('8.37')),
 (Decimal('1994'), Decimal('8.26')),
 (Decimal('1988'), Decimal('8.00')),
 (Decimal('2020'), Decimal('7.95')),
 (Decimal('2023'), Decimal('7.80')),
 (Decimal('2013'), Decimal('7.78'))]

Adjust top rated years to only include years that have 30+ reviewed games


In [79]:
%%sql

SELECT EXTRACT(YEAR FROM release_date) AS year, 
    COUNT(title) AS num_games_reviewed, 
    ROUND(AVG(critic_score),2) AS avg_critic_score
FROM dim_product dp
INNER JOIN dim_review dr
ON dp.unique_title = dr.unique_title
WHERE critic_score IS NOT NULL
GROUP BY EXTRACT(YEAR FROM release_date)
HAVING COUNT(DISTINCT title) > 30
ORDER BY avg_critic_score DESC
LIMIT 10;

 * postgresql://postgres:***@localhost:5432/Videogame_Sales
10 rows affected.


[(Decimal('2013'), 133, Decimal('7.78')),
 (Decimal('2014'), 86, Decimal('7.71')),
 (Decimal('2018'), 138, Decimal('7.71')),
 (Decimal('2017'), 129, Decimal('7.69')),
 (Decimal('1996'), 59, Decimal('7.65')),
 (Decimal('2016'), 80, Decimal('7.57')),
 (Decimal('2012'), 124, Decimal('7.51')),
 (Decimal('2015'), 83, Decimal('7.49')),
 (Decimal('1997'), 75, Decimal('7.48')),
 (Decimal('1998'), 97, Decimal('7.41'))]

Check which years were dropped due to adding the condition of at least 30 reviews


In [80]:
%%sql

WITH top_critic_years AS(
    SELECT EXTRACT(YEAR FROM release_date) AS year, 
        COUNT(title) AS num_games_reviewed,
        ROUND(AVG(critic_score), 2) AS avg_critic_score
    FROM dim_product AS dp
    LEFT JOIN dim_review AS dr
    ON dp.unique_title = dr.unique_title
    WHERE critic_score IS NOT NULL
    GROUP BY year
    ORDER BY avg_critic_score DESC
    LIMIT 10
), top_critic_years_often_reviewed AS(
    SELECT EXTRACT(YEAR FROM release_date) AS year, 
        COUNT(title) AS num_games_reviewed, 
        ROUND(AVG(critic_score),2) AS avg_critic_score
    FROM dim_product dp
    INNER JOIN dim_review dr
    ON dp.unique_title = dr.unique_title
    WHERE critic_score IS NOT NULL
    GROUP BY EXTRACT(YEAR FROM release_date)
    HAVING COUNT(DISTINCT title) > 30
    ORDER BY avg_critic_score DESC
    LIMIT 10
)
SELECT year, 
    num_games_reviewed,
    avg_critic_score
FROM top_critic_years
EXCEPT
SELECT year, 
    num_games_reviewed,
    avg_critic_score
FROM top_critic_years_often_reviewed
ORDER BY avg_critic_score DESC;

 * postgresql://postgres:***@localhost:5432/Videogame_Sales
9 rows affected.


[(Decimal('1985'), 1, Decimal('10.00')),
 (Decimal('1992'), 4, Decimal('9.13')),
 (Decimal('1987'), 1, Decimal('9.00')),
 (Decimal('1991'), 6, Decimal('8.80')),
 (Decimal('1993'), 12, Decimal('8.37')),
 (Decimal('1994'), 15, Decimal('8.26')),
 (Decimal('1988'), 1, Decimal('8.00')),
 (Decimal('2020'), 13, Decimal('7.95')),
 (Decimal('2023'), 2, Decimal('7.80'))]

The top 10 highest rated years contained several very high rated years, however on closer inspection these years had very little reviews and therefore are more likely to be outliers. Especially 1985, 1987 and 1988, all having only one review.